roborazzi-idea-plugin/src/main/kotlin/com/github/takahirom/roborazzi/idea/preview/RoborazziGradleTask.kt

```diff
@@ -0,0 +1,70 @@
+package com.github.takahirom.roborazzi.idea.preview
+
+import com.intellij.execution.executors.DefaultRunExecutor
+import com.intellij.openapi.externalSystem.model.DataNode
+import com.intellij.openapi.externalSystem.model.ProjectKeys
+import com.intellij.openapi.externalSystem.model.execution.ExternalSystemTaskExecutionSettings
+import com.intellij.openapi.externalSystem.model.project.ModuleData
+import com.intellij.openapi.externalSystem.service.execution.ProgressExecutionMode
+import com.intellij.openapi.externalSystem.task.TaskCallback
+import com.intellij.openapi.externalSystem.util.ExternalSystemApiUtil
+import com.intellij.openapi.externalSystem.util.ExternalSystemUtil
+import com.intellij.openapi.fileEditor.FileEditorManager
+import com.intellij.openapi.project.Project
+import com.intellij.psi.PsiDocumentManager
+import com.intellij.psi.PsiFile
+import org.jetbrains.kotlin.idea.util.projectStructure.module
+import org.jetbrains.kotlin.psi.KtFile
+import org.jetbrains.plugins.gradle.util.GradleConstants
+import org.jetbrains.plugins.gradle.util.GradleUtil
+
+class RoborazziGradleTask {
+
+  fun fetchTasks(project: Project): List<String> {
+    val gradleModuleData = getGradleData(project) ?: return emptyList()
+
+    return ExternalSystemApiUtil.findAll(gradleModuleData, ProjectKeys.TASK)
+      .filter { it.data.name.contains("Roborazzi", true) && it.data.name.contains("DirRoborazzi", true).not()}
+      .map { it.data.name }.sorted()
```

Comment:
I would like to refresh only the current module, as it could take a lot of time to build all modules. How about using gradleModuleData.data.id?

```suggestion
      .map { gradleModuleData.data.id + ":" + it.data.name }.sorted()
```